src/utils/tickets.js

```diff
@@ -30,26 +31,34 @@ const processTicket = async ticket => {
   const artifacts = [];
 
   // process resources, waits for page to finish loading
-  const jsArtifacts = await resourceManager.process();
-  artifacts.push(...jsArtifacts);
+  const resourceArtifacts = await resourceManager.process();
+  artifacts.push(...resourceArtifacts);
 
   // needed for safe browsing:
   const urls = resourceManager.getURLs();
 
-  // process screenshots. ocr depends on this, so doing in sync...
-  const ssArtifacts = await ssManager.processScreenshots(ticket, page);
+  // filter out only js files for yara:
+  const jsArtifacts = resourceArtifacts.filter(artifact =>
+    artifact.filename.endsWith('.js')
+  );
+
+  // filter out images for OCR:
+  const imgArtifacts = resourceArtifacts.filter(artifact =>
+    isImageUrl(artifact.filename)
+  );
 
   // do next 3 tasks in parallel using Promise.all.
   // ocr, yara, safe browsing API
   const asyncArtifacts = await Promise.all([
-    ocrManager.processImages(ssArtifacts),
+    ssManager.processScreenshots(ticket, page),
+    ocrManager.processImages(imgArtifacts),
     yaraManager.setupResourceScan(jsArtifacts, ticket),
     safeBrowsingManager.getMalwareMatches(urls),
   ]);
 
-  artifacts.push(...ssArtifacts); // screenshots
-  artifacts.push(...asyncArtifacts[0]); // ocr
-  artifacts.push(...asyncArtifacts[1]); // yara
+  artifacts.push(...asyncArtifacts[0]); // screenshots
+  artifacts.push(...asyncArtifacts[1]); // ocr
+  artifacts.push(...asyncArtifacts[2]); // yara
```

Comment:
@williamtdr looks like we need to update the malware matches in line 76 to reflect added element in asyncArtifacts. Otherwise this looks really good to me, nice work!